api/auth/cert/cert.go

```diff
@@ -0,0 +1,163 @@
+package cert
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/hashicorp/go-rootcerts"
+	"github.com/hashicorp/vault/api"
+)
+
+type CertAuth struct {
+	role               string
+	caCert             string
+	caCertBytes        []byte
+	clientCert         string
+	clientKey          string
+	insecureSkipVerify bool
+}
+
+var _ api.AuthMethod = (*CertAuth)(nil)
+
+type LoginOption func(a *CertAuth) error
+
+// NewCertAuth initializes a new cert auth method interface to be
+// passed as a parameter to the client.Auth().Login method.
+//
+// Supported options: WithClientCertAndKey, WithInsecure
+//
+// https://developer.hashicorp.com/vault/api-docs/auth/cert#login-with-tls-certificate-method
+func NewCertAuth(roleName string, opts ...LoginOption) (*CertAuth, error) {
+	a := &CertAuth{
+		role: roleName,
+	}
```

Comment:
```suggestion
	a := &CertAuth{
		role: roleName,
	}
```
if roleName == "" {
	return nil, fmt.Errorf("no role name provided for login")
}
a := &CertAuth{
		role: roleName,
	}